### PR TITLE
Default gasprice + new rpc methods

### PR DESCRIPTION
**Pull request by hackaugusto**
- geth will ignore a transaction with a low gasprice and blacklist the
user, this caused trouble because the default gasprice used in the
deploy_solidity_contract method was 1 wei.
- pooling didn't account for the removal of a transaction (that can be
caused by the low gas price)
- added new rpc methods:
  - eth_estimateGas
  - eth_getTransactionReceipt
  - eth_getCode
- added a utility for the Proxy to get the estimated gas usage.

### CODE REVIEW

**Comment by LefterisJP** [comment]
@heikoheiko Can we get this PR merged ASAP? If this PR is not in place one of the raiden integration tests fail:

`raiden/tests/integration/test_blockchainservice.py::test_new_netting_contract`

File: pyethapp/rpc_client.py [hunk 1]
```diff
@@ -538,55 +661,81 @@ def poll(self, transaction_hash, confirmations=None, timeout=None):
...
-        confirmation_block = transaction_block + confirmations
+                block_number = self.blocknumber()
 
-        block_number = self.blocknumber()
-        while confirmation_block > block_number:
-            if deadline and time.time() > deadline:
-                raise Exception('timeout when waiting for confirmation')
+                while block_number < confirmation_block:
+                    gevent.sleep(.5)
+                    block_number = self.blocknumber()
```
Comment (ezdac) [inline pyethapp/rpc_client.py:717]: `return True` or `return block_number`
at this point the transaction was successful

File: pyethapp/rpc_client.py [hunk 2]
```diff
@@ -538,55 +661,81 @@ def poll(self, transaction_hash, confirmations=None, timeout=None):
...
+            last_result = None
+
+            while True:
+                # Could return None for a short period of time, until the
+                # transaction is added to the pool
+                transaction = self.call('eth_getTransactionByHash', transaction_hash)
+
+                # if the transaction was added to the pool and then removed
+                if transaction is None and last_result is not None:
+                    raise Exception('invalid transaction, check gas price')
```
Comment (ezdac) [inline pyethapp/rpc_client.py:697]: distinguish exeptions (e.g. `raise InvalidTransaction`)

File: pyethapp/rpc_client.py [hunk 3]
```diff
@@ -538,55 +661,81 @@ def poll(self, transaction_hash, confirmations=None, timeout=None):
...
-            if deadline and time.time() > deadline:
-                raise Exception('timeout when waiting for confirmation')
+                while block_number < confirmation_block:
+                    gevent.sleep(.5)
+                    block_number = self.blocknumber()
 
-            gevent.sleep(.5)
-            block_number = self.blocknumber()
+        except gevent.Timeout:
+            raise Exception('timeout when polling for transaction')
```
Comment (ezdac) [inline pyethapp/rpc_client.py:719]: distinguish Exeptions (e.g. `raise PollTimeoutException`)

**Review by ezdac — CHANGES_REQUESTED** [review]
include return value and error distinction to `poll` method

File: pyethapp/rpc_client.py (hunk 1 above)
Comment (hackaugusto) [reply pyethapp/rpc_client.py:717]: Now returning the current block number, this will include the confirmations block.

**Review by ezdac — APPROVED** [review]
Looks good to me.

**Pull request merged**